### PR TITLE
introspection: Fix hostname having  as character

### PR DESCRIPTION
**Pull request by JafarAbdi**
Fix: https://github.com/ros-planning/moveit_task_constructor/issues/222

### CODE REVIEW

**Comment by JafarAbdi** [comment]
This's the Linux hostname specification https://man7.org/linux/man-pages/man7/hostname.7.html

**Review by felixvd — APPROVED** [review]
LGTM, although I haven't tested. Can't imagine many failure modes though.

**Review by mamoll — APPROVED** [review]
LGTM. This seems like a straightforward fix.

File: core/src/introspection.cpp [hunk 1]
```diff
@@ -57,6 +57,8 @@ std::string getTaskId(const TaskPrivate* task) {
 	std::ostringstream oss;
 	char our_hostname[256] = { 0 };
 	gethostname(our_hostname, sizeof(our_hostname) - 1);
+	// Hostname could have `-` as a character but this is an invalid character in ROS so we replace it with `_`
+	std::replace(std::begin(our_hostname), std::end(our_hostname), '-', '_');
```
Comment (JafarAbdi) [inline core/src/introspection.cpp:61]: Or maybe replacing these lines with

```cpp
static const auto our_hostname = [] {
   char our_hostname[256] = { 0 };
   gethostname(our_hostname, sizeof(our_hostname) - 1);
   std::replace(std::begin(our_hostname), std::end(our_hostname), '-', '_');
   return std::string(our_hostname);
}();
```

**Review by rhaschke — APPROVED** [review]

**Pull request merged**